docs/arcadia-users-group/20221011-developer-productivity/lesson.md

```diff
@@ -0,0 +1,172 @@
+# Increasing developer productivity with IDEs and terminal customizations
+
+In the last couple of weeks, we learned many [shell](../20220906-intro-to-shell1/lesson.md) and [git](../../workshops/20220920-intro-to-git-and-github/lesson.md) commands. This week, we'll take a step back to learn about using an integrated development environment (IDE) to develop our scripts/pipelines and customizing our terminals with shortcuts and git visualizations.
+
+## Accessing the shell for this lesson
+
+For this lesson, we need to have access to a Unix shell.
+If you're not sure how to open a terminal on your computer, see [these instructions](https://swcarpentry.github.io/shell-novice/setup.html).
+
+## Caveats
+
+- You don't really have to do any of these customizations to get good at programming. But these are designed to make your life easier.
+- A good chunk of these customizations are personal. The goal is to learn _how_ to customize and to empower you to customize as you wish.
+- A good chunk of these customizations are stylistic. As a programmar, you spend a lot of time in your terminal or text editor. I think it's important to make these tools beautiful.
+- I like bash because I've been using it for 10 years. This workshop will focus on using bash instead of zsh. But there are parallels between the two and learning to customize one will help you customize the other.
+- These type of tools can sometimes get RAM-intensive. So, beware if you have a laptop with <16GB of RAM.
```

Comment:
can you flag in parenthetical which ones are the biggest offenders? like aliases shouldn't really cost any ram.